electron-app/src/websites/artconomy/artconomy.service.ts

```diff
@@ -0,0 +1,239 @@
+import { FileSubmission } from 'src/submission/file-submission/interfaces/file-submission.interface';
+import { CancellationToken } from 'src/submission/post/cancellation/cancellation-token';
+import { FilePostData } from 'src/submission/post/interfaces/file-post-data.interface';
+import { PostResponse } from 'src/submission/post/interfaces/post-response.interface';
+import { DefaultOptions } from 'src/submission/submission-part/interfaces/default-options.interface';
+import { SubmissionPart } from 'src/submission/submission-part/interfaces/submission-part.interface';
+import { ValidationParts } from 'src/submission/validator/interfaces/validation-parts.interface';
+import FileSize from 'src/utils/filesize.util';
+import WebsiteValidator from 'src/utils/website-validator.util';
+import { LoginResponse } from '../interfaces/login-response.interface';
+import { ScalingOptions } from '../interfaces/scaling-options.interface';
+import { UsernameShortcut } from '../interfaces/username-shortcut.interface';
+import { Website } from '../website.base';
+import { ArtconomyAccountData } from './artconomy-account.interface';
+import { ArtconomyDefaultFileOptions } from './artconomy.defaults';
+import { ArtconomyOptions } from './artconomy.interface';
+import {MarkdownParser} from "src/description-parsing/markdown/markdown.parser";
+
+
+@Injectable()
+export class Artconomy extends Website {
+  readonly BASE_URL: string = 'https://artconomy.com';
+  readonly acceptsFiles: string[] = [
+    'png',
+    'jpeg',
+    'jpg',
+    'gif',
+    'mp4',
+    'doc',
+    'rtf',
+    'txt',
+    'mp3',
+  ];
+  readonly defaultDescriptionParser = MarkdownParser.parse;
+  readonly fileSubmissionOptions: object = ArtconomyDefaultFileOptions;
+  usernameShortcuts: UsernameShortcut[] = [
+    {
+      key: 'ac',
+      url: 'https://artconomy/$1',
+    },
+  ];
+
+  async checkLoginStatus(data: UserAccountEntity): Promise<LoginResponse> {
+    const status: LoginResponse = { loggedIn: false, username: null};
+    if (!data.data && data.data.authtoken) {
+      return status
+    }
+    const authCheck = await Http.get<any>(`${this.BASE_URL}/api/profiles/v1/data/requester/`, data._id, {
+      requestOptions: { json: true, headers: {Authorization: `Token ${data.data.authtoken}`} },
+    });
+    if (authCheck.response.statusCode === 200 && authCheck.response.body.username !== '_') {
+      status.username = authCheck.response.body.username;
+      status.loggedIn = true;
+    }
+    return status;
+  }
+
+  transformAccountData(data: ArtconomyAccountData) {
+    return { username: data.username, id: data.id, csrftoken: data.csrftoken, authtoken: data.authtoken };
+  }
+
+  getScalingOptions(file: FileRecord): ScalingOptions | undefined {
+    return { maxSize: FileSize.MBtoBytes(200) };
+  }
+
+  preparseDescription(text: string) {
+    text = UsernameParser.replaceText(text, 'fa', '[fa]$1[/fa]');
```

Comment:
This block seems to be copy/pasted from inkbunny code. Is this intentional?
Looks like you are using markdown as your text parser but this is bbcode tagging.